text/0000-deprecate-http-tx.md

```diff
@@ -0,0 +1,123 @@
+  - The ecosystem can converge around a universally accepted transaction flow that is privacy preserving
+
+The overall expected outcome is a privacy improvement for users that may otherwise have to reveal their IP address to send or receive Grin.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+Deprecating HTTP(S) means that the old method of sending and listening for transactions over traditional HTTP(S) is no longer supported to help preserve privacy for Grin users who may not be aware of the risk to privacy using HTTP(S) poses.
+
+This reduces the overall number of transaction transport methods that need to be supported but requires that all services and wallets that previously only accepted/sent HTTP(S) transactions to upgrade at least to the version this RFC is implemented in to support their users.
+
+For the average user this means that to send or receive Grin they can use the [Slatepack](https://github.com/mimblewimble/grin-rfcs/blob/master/text/0015-slatepack.md) method. It is no longer possible to send and receive Grin via the HTTP(S) method without custom configuration.
+
+For wallet developers, it is no longer necessary to be concerned about managing HTTP(S) endpoints. All transactions will be conducted via Tor or asynchronous exchange of armored slate strings with the Slatepack standard. It can no longer be assumed that the counterparty in the transaction supports HTTP(S).
+
+Previously, some services like exchanges _only_ supported HTTP(S) Grin transactions. This change impacts these services significantly.
+
+Slatepack with Tor is the closest replacement to HTTP(S), but for some services the use of Tor is unacceptable, not possible or not allowed. These services will will need to refactor the user flow to allow supporting manual copying and pasting of [Slatepack armored transaction messages](https://github.com/mimblewimble/grin-rfcs/blob/master/text/0015-slatepack.md#slatepackmessage) instead, as the HTTP(S) method is no longer supported by default for `grin-wallet`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The technical and implementation details for this RFC are simple: no new features are added and some existing features are restricted:
+
+- `grin-wallet` http listener only accepts connections from localhost
+
+- `grin-wallet listen` no longer accepts `-e` argument
+
+- `grin-wallet send -d` provides a strong warning if the destination is not a valid Slatepack address
+
+## Deprecation Timeline
+
+- With v4.0.0 a public announcement will be made to notify the Grin ecosystem that HTTP(S) will be fully deprecated in v5.0.0
+
+- With v4.0.0 wallets will warn users that their privacy may be at risk when sending or listening via HTTP(S) and that the method will be deprecated in next major release
+
+- With v5.0.0 wallets will warn users when sending transactions via HTTP(S)
```

Comment:
> With v5.0.0 wallets will warn users when sending transactions via HTTP(S)

Warn users? My understanding was that it would not be supported at all, is that wrong?

Reply:
It's possible to technically leave the ability with some hacking/hoops but not actually support the method. I'm not opposed to removing the ability completely, but some folks may be using very custom setups and still want to use them even if traditional user-facing is no longer possible. This probably depends what @yeastplume has in mind, it may make sense to remove some methods that were previously needed to make it work. I guess if we are completely deprecating, we probably don't want to have to worry about maintaining any of these "under the hood" methods that aren't strictly necessary.